terraform/modules/jumpbox/main.tf

```diff
@@ -0,0 +1,105 @@
+## Variables
+
+variable "subnet_id" {
+  description = "The subnet which the jumpbox will be added to"
+}
+
+variable "vpc_id" {
+  description = "The VPC ID to use on this service"
+}
+
+variable "ecs_optimised_ami_version" {
+  default = "2018.03.a"
+}
+
+variable "stack_name" {
+  description = "The name of the stack the jumpbox belongs to"
+}
+
+variable "aws_region" {
+  description = "The region which the jump box will be built within"
+}
+
+variable "security_groups" {
+  type        = "list"
+  description = "A list of security groups to extend"
+  default     = []
+}
+
+variable "allowed_cidrs" {
+  type        = "list"
+  description = "List of CIDRs which are able to access the jumpbox, default are GDS ips"
+
+  default = [
+    "213.86.153.212/32",
+    "213.86.153.213/32",
+    "213.86.153.214/32",
+    "213.86.153.235/32",
+    "213.86.153.236/32",
+    "213.86.153.237/32",
+    "85.133.67.244/32",
+  ]
+}
+
+## Resources
+
+module "ssh_key" {
+  source     = "../common/ssh_key"
+  stack_name = "${var.stack_name}"
+}
+
+module "ami" {
+  source         = "../common/ami"
+  stack_name     = "${var.stack_name}"
+  amazon_release = "${var.ecs_optimised_ami_version}"
+}
+
+resource "aws_security_group" "bastion_security_group" {
+  name        = "${var.stack_name}-bastion-sg"
+  vpc_id      = "${var.vpc_id}"
+  description = "Controls access to & from the bastion"
+}
+
+resource "aws_security_group_rule" "allow_ssh" {
+  security_group_id = "${aws_security_group.bastion_security_group.id}"
+
+  type        = "ingress"
+  from_port   = 22
+  to_port     = 22
+  protocol    = "tcp"
+  cidr_blocks = ["${var.allowed_cidrs}"]
+}
+
+resource "aws_security_group_rule" "allow_all_egress" {
+  security_group_id = "${aws_security_group.bastion_security_group.id}"
+
+  type        = "egress"
+  from_port   = 0
+  to_port     = 0
+  protocol    = "-1"
+  cidr_blocks = ["${var.allowed_cidrs}"]
+}
+
+module "jump" {
```

Comment:
I wonder if a more appropriate name for this module would be `jumpbox_instance`?

Reply:
ok